### ISSUE TITLE
Video is classified as "Invalid date"

### ISSUE THREAD

**Issue by baxerus** [thumbsup/thumbsup#94]
## Bug report

If running as a Docker container:

```
Thumbsup image tag: thumbsupgallery/thumbsup           latest              17aa8336028c
Docker version: Docker version 17.12.1-ce, build 7390fc6
Operating system: Ubuntu 16.04.4 LTS
```

I'm still in testing stage. But I added a video from my iPhone. It was converted (codec copy) from MOV to MP4. This video is always classified as "Invalid date". I tried enabling DEBUG mode, getting (a little bit) more info, but nothing useful. I also mapped localtime into the docker container (was unlikely but I tried anyway) without success.

Is there a way to get more info? How is the date from the video parsed? By moment.js? I could add some logs into the code and rerun it again. But I would like to have advice first.

Best regards,

Benedict

**Comment by rprieto** [comment]
Hi Benedict, thanks for raising the issue. By classified, do you mean that you use `--albums-from date` and the name of the album is `Invalid date`?

The date is derived from a few things
- any embedded metadata
- the file name, parsed with moment.js
- the file modification date

Can you please:

- try running `exiftool <filepath> | grep Date` and share the output? There should be an entry called `DateTimeOriginal` or `CreationDate` or `CreateDate`
- share the filename, in case it's tripping up moment.js

**Comment by baxerus** [comment]
Yes, you are right, I use `--albums-from date` and it's not classified as, but sorted under the album name `Invalid date`.

The full command I am running the docker container in the moment is:
`docker run -t -v /etc/localtime:/etc/localtime:ro -v $(pwd)/attachements:/input -v $(pwd)/thumbsup:/output thumbsupgallery/thumbsup thumbsup --input /input --output /output --cleanup true --albums-from date --theme cards --download-photos copy --download-videos copy --thumb-size 200 --large-size 1080 --albums-date-format "YYYY/MM.YYYY"`

The filename of the file I have the problem with is `IMG_0526.mp4`, so no date in it (if not 0526 is mistaken as one).

I changed the docker command to:
`docker run -t -v $(pwd)/attachements:/input thumbsupgallery/thumbsup exiftool /input/IMG_0526.mp4 | grep Date`

and get this output:

```File Modification Date/Time     : 2018:03:21 13:09:18+00:00
File Access Date/Time           : 2018:03:21 13:09:45+00:00
File Inode Change Date/Time     : 2018:03:21 13:09:18+00:00
Create Date                     : 0000:00:00 00:00:00
Modify Date                     : 0000:00:00 00:00:00
Track Create Date               : 0000:00:00 00:00:00
Track Modify Date               : 0000:00:00 00:00:00
Media Create Date               : 0000:00:00 00:00:00
Media Modify Date               : 0000:00:00 00:00:00
```

and with `ffprobe` at the end the output is:

```Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/input/IMG_0526.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf57.56.101
  Duration: 00:00:08.26, start: 0.000000, bitrate: 23393 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709), 1920x1080, 23282 kb/s, 59.95 fps, 59.94 tbr, 19200 tbn, 38400 tbc (default)
    Metadata:
      rotate          : 90
      handler_name    : VideoHandler
    Side data:
      displaymatrix: rotation of -90.00 degrees
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 100 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
```

So there is really is no `Date` metadata in the file. Shouldn't it fallback to the modification date of the file than? BTW Do you mean the one in the filesystem or form the EXIF data?

**Comment by baxerus** [comment]
Okay, I "fixed" the problem for my use case upstream before `thumbsup` by mapping the metadata into the MP4 files.

But still I would assume there should be a fallback inside from `thumbsup` in place, for the case that `exiftool` only returns `0000:00:00 00:00:00`. Even falling back to NOW (or file creation date form the filesystem) would be better than `Invalid date`, wouldn't it?

**Comment by rprieto** [comment]
Hi, thanks a lot for all this troubleshooting. You’re right you found the issue, for now thumbsup assumes that if Exiftool returns a date then it must be valid. I had never seen `0000:00:00 00:00:00` until now.

I’ll add a check to make sure the date is valid shortly (`moment.isValid`). Then it will automatically fall back to using the file name (when applicable) or file modification date if needed.


Thanks!
